### PR TITLE
Fixing stackoverflow error on surv.glmboost

### PR DESCRIPTION
**Pull request by vlegoff**
Hello,

while using surv.glmboost on tasks with large number of features (~1e5), I got a stackoverflow error identical to the one documented here: https://github.com/mlr-org/mlr/issues/1738

This PR aims to fix this error by avoiding the formula interface for surv.glmboost

### CODE REVIEW

**Comment by bblodfon** [comment]
@vlegoff thanks! Actually I remember testing this myself one years ago and have the note "very slow + memory footprint of trained model huge" - maybe this fixes things! Also, since we are here, we can implement feature selection with `model$xselect()` and variable importance with `mboost::varimp()`, can you maybe confirm, have you tried this? (again from my notes one year ago)

**Comment by sebffischer** [comment]
@bblodfon do you think we can merge this?

**Comment by jemus42** [comment]
For what it's worth, I remember from talking to Marvin (maintainer of `ranger`) that the formula interface is known to be a bad choice in the case of very high numbers of features for memory reasons.
From my understanding that implies that in general, for any given learner, a data.frame/matrix interface should be preferred over the formula interface as the safer default choice.

I haven't checked but maybe there are other learners where it makes sense to switch away from the formula interface?

**Comment by bblodfon** [comment]
Added the `matrix` interface (it is truly faster) and added support for `importance()` and `selected_features()` when the train `task` desn't have `factor` variables.

File: R/learner_mboost_surv_glmboost.R [hunk 1]
```diff
@@ -47,7 +57,7 @@ LearnerSurvGLMBoost = R6Class("LearnerSurvGLMBoost",
         contrasts.arg = p_uty(tags = "train")
       )
 
-      ps$values = list(family = "coxph")
+      ps$values = list(family = "coxph", center = FALSE)
```
Comment (sebffischer) [inline R/learner_mboost_surv_glmboost.R:60]: why was this changed? Such changes must always be documented in the learner description and we should give an explanation why this is the case

Comment (sebffischer) [reply R/learner_mboost_surv_glmboost.R:60]: if it is deprecated as you write below we should maybe just remove the parameter, but maybe I don't understand it properly yet.

File: tests/testthat/test_mboost_surv_glmboost.R [hunk 2]
```diff
@@ -1,15 +1,26 @@
 test_that("autotest ph", {
   learner = lrn("surv.glmboost", family = "coxph")
   expect_learner(learner)
+  exclude_str = "weights|utf8_feature_names|feat_single_factor|feat_all"
   # weights are fine for all predict types except 'distr'
-  result = run_autotest(learner, exclude = "weights|utf8_feature_names", check_replicable = FALSE)
```
Comment (sebffischer) [inline tests/testthat/test_mboost_surv_glmboost.R:5]: why do we now have to exclude more tests?

**Review by sebffischer — CHANGES_REQUESTED** [review]

File: R/learner_mboost_surv_glmboost.R (hunk 1 above)
Comment (bblodfon) [reply R/learner_mboost_surv_glmboost.R:60]: Went through a bit more deeply in the documentation, it's actually `center` = `TRUE` by default:
image

File: tests/testthat/test_mboost_surv_glmboost.R (hunk 2 above)
Comment (bblodfon) [reply tests/testthat/test_mboost_surv_glmboost.R:5]: The `importance` and `selected_features` functions don't work for datasets with factor features, I have a `stop` function there. But I added a extra test with such a dataset that has many factors

File: R/learner_mboost_surv_glmboost.R (hunk 1 above)
Comment (bblodfon) [reply R/learner_mboost_surv_glmboost.R:60]: the other thing is that `center` it is part of the control variables but deprecated but they also have it right now in the main `glmboost` function (but since it is still accessible from the `mboost_control` so we need to remove it from there):
image

Comment (bblodfon) [reply R/learner_mboost_surv_glmboost.R]: @sebffischer summary: it's a code-related thing and I needed to do that since they haven't completely deprecated `center` from the `mboost::mboost_control()` function (it's there + in the main function as parameter).

Also, I reverted to `center = TRUE` as it was before

Comment (sebffischer) [reply R/learner_mboost_surv_glmboost.R]: ok great!
